package main

import (
	"errors"
	"fmt"
)

func primosAteN(n int) ([]int, error) {
	if n < 0 {
		return nil, errors.New("Número negativo. Não é possível calcular primos.")
	}

	var primos []int

	for i := 2; i <= n; i++ {
		if ehPrimo(i) {
			primos = append(primos, i)
		}
	}

	return primos, nil
}

func ehPrimo(numero int) bool {
	if numero < 2 {
		return false
	}

	limite := int(numero / 2)

	for i := 2; i <= limite; i++ {
		if numero%i == 0 {
			return false
		}
	}

	return true
}
func main() {

numero := 20
 
 
 resultado, err := primosAteN(numero)

	if err != nil {
		fmt.Println("Erro:", err)
	} else {
		fmt.Printf("Números primos até %d: %v\n", numero, resultado)
	}
}

